src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/DataGridViewTests.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Data;
+using System.Drawing;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.Windows.Forms.UITests
+{
+    public class DataGridViewTests : ControlTestBase
+    {
+        public DataGridViewTests(ITestOutputHelper testOutputHelper)
+            : base(testOutputHelper)
+        {
+        }
+
+        [WinFormsFact]
+        public async Task DataGridView_ToolTip_DoesNot_ThrowExceptionAsync()
+        {
+            await RunTestAsync(async (form, dataGridView) =>
+            {
+                using DataTable dataTable = new();
+                dataTable.Columns.Add(columnName: "name");
+                dataTable.Rows.Add(values: "name1");
+                dataGridView.ShowCellToolTips = true;
+                dataGridView.DataSource = dataTable;
+                Point point = dataGridView.GetCellDisplayRectangle(columnIndex: 0, rowIndex: 0, cutOverflow: false).Location;
+
+                // Move mouse cursor over any cell of the first row.
+                await InputSimulator.SendAsync(
+                            form,
+                            inputSimulator => inputSimulator.Mouse.MoveMouseTo(point.X, point.Y));
+
+                // We need to close the form and then check whether dataTable works fine or it doesn't
+                form.Close();
+                var exceptionThrown = false;
+                try
+                {
+                    dataTable.AcceptChanges();
+                }
+                catch
+                {
+                    exceptionThrown = true;
+                }
+
+                Assert.False(exceptionThrown);
```

Comment:
```suggestion
                dataTable.AcceptChanges();
```

If there's an exception - the test will fail.